commands.go

```diff
@@ -852,6 +860,30 @@ func (cmd *Part5Supported) SetValue(result *Result) error {
 	return nil
 }
 
+type ClearTroubleCodes struct {
+	BaseCommand
+	ResultLessCommand
+}
+
+func (cmd *ResultLessCommand) SetValue(result *Result) error {
+	return nil
+}
+func (cmd *ResultLessCommand) ValueAsLit() string {
+	return ""
+}
+
+func NewService04Command(parameterID byte, dataWidth byte, key string) BaseCommand {
```

Comment:
I don't think hiding one argument justifies the need to introduce a new helper. 

Also it's not consistent with how service 01 commands are created, they don't have a `NewService01Command` helper.

Reply:
You're absolutely right. Running a ServiceXXCommand  doesn't make sense (as opposed to running a more concrete command like "EngineRPM".  Sorry for that. But this makes me think about something: Should we change BaseCommand to  baseCommand (not  exported) since it  wouldn't make sense for a user to run a  "raw" base command? It would kind of force users/devs to add  the command if it's missing.

Reply:
Yes, great suggestion! 

I think that will make it more straightforward for users of the library how you use the commands and more clear for the developers of the library what role `BaseCommand` has. 

It also aligns with the goal of the library to provide concrete types for all the standard OBD-II PIDs (as defined by SAE J1979). 

If people want to create custom commands that are not in the standard, they can still create their own concrete types that implement the `OBDCommand` interface.